packages/eslint-plugin/src/stylex-sort-keys.js

```diff
@@ -0,0 +1,292 @@
+  }
+
+  return prevName <= currName;
+}
+
+function getPropertyName(node: Property): string | null {
+  const staticName = getStaticPropertyName(node);
+
+  return staticName !== null ? staticName : node.key.name || null;
+}
+
+const stylexSortKeys = {
+  type: 'suggestion',
+  docs: {
+    description: 'Require style properties to be sorted by key',
+    recommended: false,
+    url: 'https://github.com/facebook/stylex/tree/main/packages/eslint-plugin',
+  },
+  schema: [
+    {
+      type: 'object',
+      properties: {
+        validImports: {
+          type: 'array',
+          items: { type: 'string' },
+          default: ['stylex', '@stylexjs/stylex'],
+        },
+        minKeys: {
+          type: 'integer',
+          minimum: 2,
+          default: 2,
+        },
+        allowLineSeparatedGroups: {
+          type: 'boolean',
+          default: false,
+        },
+      },
+      additionalProperties: false,
+    },
+  ],
+  create(context: Rule.RuleContext): { ... } {
+    const {
+      validImports: importsToLookFor = ['stylex', '@stylexjs/stylex'],
+      minKeys = 2,
+      allowLineSeparatedGroups = false,
+    }: Schema = context.options[0] || {};
+
+    const styleXDefaultImports = new Set<string>();
+    const styleXCreateImports = new Set<string>();
+
+    function isStylexCallee(node: Node) {
+      return (
+        (node.type === 'MemberExpression' &&
+          node.object.type === 'Identifier' &&
+          styleXDefaultImports.has(node.object.name) &&
+          node.property.type === 'Identifier' &&
+          node.property.name === 'create') ||
+        (node.type === 'Identifier' && styleXCreateImports.has(node.name))
+      );
+    }
+
+    function isStylexDeclaration(node: $ReadOnly<{ ...Node, ... }>) {
+      return (
+        node &&
+        node.type === 'CallExpression' &&
+        isStylexCallee(node.callee) &&
+        node.arguments.length === 1
+      );
+    }
+
+    let stack: Stack = null;
+    let isInsideStyleXCreateCall = false;
+    let objectExpressionNestingLevel = -1;
+
+    return {
+      ImportDeclaration(node: ImportDeclaration) {
+        if (
+          node.source.type !== 'Literal' ||
+          typeof node.source.value !== 'string'
+        ) {
+          return;
+        }
+
+        if (!importsToLookFor.includes(node.source.value)) {
+          return;
+        }
+
+        node.specifiers.forEach((specifier) => {
+          if (
+            specifier.type === 'ImportDefaultSpecifier' ||
+            specifier.type === 'ImportNamespaceSpecifier'
+          ) {
+            styleXDefaultImports.add(specifier.local.name);
+          }
+
+          if (
+            specifier.type === 'ImportSpecifier' &&
+            specifier.imported.name === 'create'
+          ) {
+            styleXCreateImports.add(specifier.local.name);
+          }
+        });
+      },
+      CallExpression(node: { ...CallExpression, ...Rule.NodeParentExtension }) {
```

Comment:
Make sure to wrap every object type with `$ReadOnly`. This will prevent most of the type errors that tend to trip people up.